controllers/apim/gateway_rlp_eventmapper.go

```diff
@@ -0,0 +1,58 @@
+package apim
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	apimv1alpha1 "github.com/kuadrant/kuadrant-controller/apis/apim/v1alpha1"
+	"github.com/kuadrant/kuadrant-controller/pkg/rlptools"
+)
+
+// GatewayEventMapper is an EventHandler that maps Gateway object events to policy events.
```

Comment:
`GatewayRateLimitPolicyEventMapper` *

Reply:
Maybe: GatewayRateLimitPolicyEventMapper is an EventHandler that maps (Gateway) RLP object events to (Route) RLP events (?)

Reply:
I am afraid, not only Route RLP events, Gateway RLP can also be requested.

Changed to
```
// GatewayRateLimitPolicyEventMapper is an EventHandler that maps:             
// RLP events targeting a Gateway TO all the RLPs configuring that same gateway
```